File: shared_flutter/lib/src/api/models/retailer_sync_response.dart
import 'retailer_list_item.dart';

class RetailerSyncResponse {
  RetailerSyncResponse({required this.cursor, required this.changes, required this.deletes});

  final String cursor;
  final List<RetailerListItem> changes;
  final List<String> deletes;

  factory RetailerSyncResponse.fromJson(Map<String, dynamic> json) => RetailerSyncResponse(
        cursor: json['cursor'] as String,
        changes: ((json['changes'] as List?) ?? const [])
            .map((e) => RetailerListItem.fromJson(e as Map<String, dynamic>))
            .toList(),
        deletes: ((json['deletes'] as List?) ?? const []).map((e) => e as String).toList(),
      );
}

File: backend/src/GasFinder.Shared/Contracts/Sync/RetailerSyncResponse.cs
using GasFinder.Shared.Contracts.Retailers;

namespace GasFinder.Shared.Contracts.Sync;

public record RetailerSyncResponse(
    string Cursor,
    IReadOnlyList<RetailerListItem> Changes,
    IReadOnlyList<Guid> Deletes
);
